autox/informer_test.go

```diff
@@ -1,28 +1,152 @@
 package autox
 
+// import (
+// 	"testing"
+
+// 	"github.com/stretchr/testify/assert"
+
+// 	"helm.sh/helm/v3/pkg/cli"
+// 	"k8s.io/apimachinery/pkg/runtime/schema"
+// )
+
+// func TestNewInformer(t *testing.T) {
+// 	Client = *newFakeKubeClient(cli.New())
+// 	w := newIter8Watcher(
+// 		[]schema.GroupVersionResource{{
+// 			Group:    "",
+// 			Version:  "v1",
+// 			Resource: "services",
+// 		}, {
+// 			Group:    "apps",
+// 			Version:  "v1",
+// 			Resource: "deployments",
+// 		}},
+// 		[]string{"default", "foo"},
+// 		chartGroupConfig{},
+// 	)
+// 	assert.NotNil(t, w)
+// }
+
 import (
+	"context"
 	"testing"
+	"time"
 
-	"github.com/stretchr/testify/assert"
+	// abnapp "github.com/iter8-tools/iter8/abn/application"
+	// "github.com/iter8-tools/iter8/abn/k8sclient"
 
+	"github.com/stretchr/testify/assert"
 	"helm.sh/helm/v3/pkg/cli"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func TestNewInformer(t *testing.T) {
+var TRACK_ANNOTATION = "iter8.tools/track"
+var NAME_LABEL = "app.kubernetes.io/name"
+var VERSION_LABEL = "app.kubernetes.io/version"
+var READY_ANNOTATION = "iter8.tools/ready"
+var ITER8_ANNOTATION = "iter8.tools/abn"
+var ITER8_LABEL = "iter8.tools/abn"
+
+func TestInformer(t *testing.T) {
+	gvr := schema.GroupVersionResource{
+		Group:    "apps",
+		Version:  "v1",
+		Resource: "deployments",
+	}
+	namespace := "default"
+	application := "demo"
+	version := "v1"
+	track := ""
+
+	// define and start watcher
 	Client = *newFakeKubeClient(cli.New())
```

Comment:
Usually `*object` is unnecessary and not recommended. 

The pattern we generally want to shoot for is this: https://go.dev/tour/methods/8

Of course, there might be exceptions: https://flaviocopes.com/golang-methods-receivers/